lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -276,7 +276,13 @@ def main():
     # find the group if present
     group = None
     groups = {}
-    for curGroup in ec2.get_all_security_groups():
+
+    try:
+        security_groups = ec2.get_all_security_groups()
+    except Exception as e:
```

Comment:
@willthames @s-hertel is this a request to upgrade boto to boto3 for ec2_group?  happy to take a crack at that (trying now), but a bit larger than the original scope of this PR